slider/tests/test_beatmap.py

```diff
@@ -237,3 +237,66 @@ def test_hp(beatmap):
 
 def test_od(beatmap):
     assert beatmap.od() == 9
+
+
+def test_pack(beatmap):
+    # Pack the beatmap and parse it again to see
+    # if there is difference.
+    beatmap = slider.example_data.beatmaps.miiro_vs_ai_no_scenario('Tatoe')
+    packed_str = beatmap.pack()
+    # with open(r'.\test.osu', 'wt') as f:
+    #     f.write(packed_str)
+    packed_beatmap = slider.Beatmap.parse(packed_str)
+    # Since sections like Colours and Events are currently omitted by
+    # ``Beatmap.parse``, these sections will be missing in .osu files
+    # written back from parsed Beatmaps. Fortunately, without these
+    # sections, rewritten .osu can still be recognized by osu! client.
+    for field, field_value in beatmap.__dict__.items():
+        assert field in packed_beatmap.__dict__
+        if field == 'timing_points':
+            # Check if all timing points are the same.
+            for timing_point, packed_timing_point in \
+                    zip(field_value, packed_beatmap.timing_points):
+                for timing_point_field, timing_point_field_value in \
+                        timing_point.__dict__.items():
+                    # inherited timing point has field parent, skip it
+                    if not isinstance(timing_point_field_value,
+                                      slider.beatmap.TimingPoint):
+                        assert timing_point_field in \
+                               packed_timing_point.__dict__
+                        assert timing_point_field_value == \
+                            packed_timing_point.__dict__[timing_point_field]
+        elif field == '_hit_objects':
+            # Check if all hit objects are the same.
+            for hit_object, packed_hit_object in \
+                    zip(beatmap.hit_objects(), packed_beatmap.hit_objects()):
+                for hit_object_field, hit_object_field_value in \
+                        hit_object.__dict__.items():
```

Comment:
similarly here, accessing `__dict__` makes this code harder to read, not easier. Just explicitly assert that the fields you want to be equal are equal.